scripts/migration/migrate-to-asm

```diff
@@ -1337,13 +1495,90 @@ apply_backup() {
 }
 
 remove_asm() {
-  run istioctl x uninstall --revision "${REVISION_LABEL}" -y
+  run istioctl x uninstall --revision "$(asm_revision_label)" -y
   run kubectl delete ns asm-system
 
   info "****"
   info "Anthos Service Mesh has been uninstalled from the cluster."
 }
 
+should_use_mcp() {
+  if [[ "${NO_MCP}" -eq 1 ]]; then false; return; fi
+  if [[ -f ./configuration_backup/use_mcp ]]; then return; fi
+  if [[ -f ./configuration_backup/no_use_mcp ]]; then false; return; fi
+
+  run_mcp_fit_assessment
+}
+
+is_private_cluster() {
+  gcloud container clusters describe \
+    --region "${CLUSTER_LOCATION}" \
+    --project "${PROJECT_ID}" \
+    "${CLUSTER_NAME}" --format json \
+    | grep -q '"enablePrivateEndpoint": true,'
+}
+
+is_autopilot() {
+  gcloud container clusters describe \
+    --region "${CLUSTER_LOCATION}" \
+    --project "${PROJECT_ID}" \
+    "${CLUSTER_NAME}" --format json \
+    | grep -q 'autopilot'
+}
+
+is_supported_location() {
+  while read -r region; do
+    if [[ "${CLUSTER_LOCATION}" = "${region}"* ]]; then
+      return
+    fi
+  done <<< "$(supported_mcp_regions)"
+  false
+}
+
+run_mcp_fit_assessment() {
+  echo "Analyzing cluster configuration to determine optimal control plane..."
+
+  if ! is_supported_location; then false; return; fi
+  if is_private_cluster; then false; return; fi
+  if is_autopilot; then false; return; fi
+
+  if [[ ! -f ./configuration_backup/default_iop ]]; then
+    istioctl profile dump >./configuration_backup/default_iop 2>/dev/null
+  fi
+
+  local ARGS; ARGS="-f ./configuration_backup/default_iop"
+  if [[ -f ./configuration_backup/iop_backup ]]; then
+    ARGS="-f ./configuration_backup/iop_backup"
```

Comment:
is there only one CUSTOM_OVERLAY allowed? I thought I had seen folks passing multiple?

Reply:
For this tool only one. Do Istio users generally keep their custom IOPs separate?

Reply:
I think https://cloud.google.com/service-mesh/docs/enable-optional-features recommends it. I have seen it used a few times line that doc

Reply:
Ah, I forgot we changed this to also target ASM customers and not just OSS. I'll see how to update this.